requirements_tools/check_requirements.py

```diff
@@ -47,6 +47,9 @@ def get_raw_requirements(filename):
         (parse_requirement(line), filename)
         for line in lines
         if not line.startswith('-e ')
+        # Not just 'http' because that's a valid package name, look for urls
+        and not line.startswith('https://')
+        and not line.startswith('http://')
```

Comment:
This has a few problems:
- the biggest being that `requirements-tools` cannot trace the transitive dependencies of such urls
- the less likely is that an http url is possibly a mutable package
- the third is that urls can't appropriately be cached by pip / pip-faster

requirements-tools is _intentionally_ strict such that if you pass its checks, you're generally doing the right thing.

I actually think the line above it (where `-e` is allowed) should also be removed since it potentially allows exactly the same problem.  probably going to make a PR which instead only allows lines that are `-e .` (generally harmless) instead of blanket allowing all editable requirements.

Reply:
Gotcha, I misunderstood the intention of the `-e ` exception to mean "ignore things we can't fully parse or check"

Based on the fact that the packages URLs cannot be checked, are you two of the opinion that URLs should never be used in a requirements.txt and that `requirements-tools` should break when URLs are present?

Admittedly I didn't then take the new `check-requirements` that I wrote and run it on my project to see what it output. Perhaps it complaining about the URLs is exactly what's intended, but it looked to me like it was failing in some "gathering requirements" step and didn't get to do all of its checks. In which case I submit to you that if "URLs are bad", perhaps we should find a way to make them fail in a first class way and to let check-requirements run over the rest of my stuff :) 

You have a cool tool and I would like to use it to get as much info as possible, and removing the github links in our requirements is in progress but not complete , and so we're kinda stuck at the moment.

Reply:
yep totally understand.  It is intentional that it breaks in this way, however the messaging could do with some improvements -- I attempted to make it explicit and helpful in #26 -- let me know what you think of that patch :)